.github/workflows/zombienet-reusable-preflight.yml

```diff
@@ -142,4 +187,61 @@ jobs:
         shell: bash
         run: |
           echo "workflow file: ${{ steps.current_file.outputs.currentWorkflowFile }}"
-          echo "Modified: ${{ steps.set_changes.outputs.modified_keys }}"
\ No newline at end of file
+          echo "Modified: ${{ steps.set_changes.outputs.modified_keys }}"
+          echo "ZOMBIENET_IMAGE: ${{ steps.set_vars.outputs.ZOMBIENET_IMAGE }}"
+
+  #
+  #
+  #
+  ci-env:
+    needs: [preflight]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Info vars
+        run: |
+          echo "ZOMBIENET_IMAGE: ${{ needs.preflight.outputs.ZOMBIENET_IMAGE }}"
+          echo "ZOMBIENET_RUNNER: ${{ needs.preflight.outputs.ZOMBIENET_RUNNER }}"
+          echo "DOCKER_IMAGES_VERSION: ${{ needs.preflight.outputs.DOCKER_IMAGES_VERSION }}"
+          echo "SOURCE_REF_SLUG: ${{ needs.preflight.outputs.SOURCE_REF_SLUG }}"
+          echo "PUSHGATEWAY_URL: ${{ needs.preflight.outputs.PUSHGATEWAY_URL }}"
+          echo "DEBUG: ${{ needs.preflight.outputs.DEBUG }}"
+          echo "ZOMBIE_PROVIDER: ${{ needs.preflight.outputs.ZOMBIE_PROVIDER }}"
+          echo "RUST_LOG: ${{ needs.preflight.outputs.RUST_LOG }}"
+          echo "RUN_IN_CI: ${{ needs.preflight.outputs.RUN_IN_CI }}"
+          echo "KUBERNETES_CPU_REQUEST: ${{ needs.preflight.outputs.KUBERNETES_CPU_REQUEST }}"
+          echo "KUBERNETES_MEMORY_REQUEST: ${{ needs.preflight.outputs.KUBERNETES_MEMORY_REQUEST }}"
+          #
+          echo "github.ref: ${{ github.ref }}"
+          echo "github.ref_name: ${{ github.ref_name }}"
+          echo "github.sha: ${{ github.sha }}"
+  #
+  #
+  # Wait until the 'build and push' image workflow success
+  wait_build_images:
+    needs: [ci-env]
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Wait until "Build and push images" workflow is done
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          gh --version
+          export SHA=${{ github.event.pull_request.head.sha }}
+          while true; do
+            STATUS=$(gh run ls -c $SHA -w "Build and push images" --json name,status --jq '.[] | select(.name == "Build and push images") | .status')
```

Comment:
This is good as a temporary solution for now but in the future it would be nice to add an exception here. I'm pretty sure that we'll face the situation when gh cli tool will receive some 4xx or 5xx error from github api as it happened yesterday (06.02.25)

Reply:
Yes, make sense. From yesterday, when this command get an errro from `github` this job fails. Maybe we can keep retrying on `4xx` but not on `5xx`.